crates/polars-ops/src/frame/join/iejoin/mod.rs

```diff
@@ -364,6 +364,15 @@ unsafe fn materialize_join(
     suffix: Option<PlSmallStr>,
 ) -> PolarsResult<DataFrame> {
     try_raise_keyboard_interrupt();
+    let mut sorted_left_row_idx = left_row_idx.sort(false).into_no_null_iter().peekable();
+    let left_idx_not_in_inner = (0..left.height() as u64).into_iter().filter(|left| {
+        while sorted_left_row_idx.next_if(|inner| inner < left).is_some() {}
+        sorted_left_row_idx.peek().map_or(true, |inner| left < inner)
+    });
+    let outer_left_row_idx: IdxCa = IdxCa::from_iter(left_idx_not_in_inner);
+    let right_nulls = outer_left_row_idx.clear() as IdxCa;
+    let left_row_idx = &(left_row_idx + outer_left_row_idx) as &IdxCa;
+    let right_row_idx = &(right_row_idx + right_nulls) as &IdxCa;
```

Comment:
Polars implements the `Add` trait for two `ChunkedArray`s or two `&ChunkedArray`, but not adding a `ChunkedArray` to an `&ChunkedArray`:
```suggestion
    let left_row_idx = &(left_row_idx + &outer_left_row_idx) as &IdxCa;
    let right_row_idx = &(right_row_idx + &right_nulls) as &IdxCa;
```